src/matrixroomhandler.ts

```diff
@@ -74,19 +91,178 @@ export class MatrixRoomHandler {
+          return this.bridge.getIntent().sendMessage(event.room_id, {
+              msgtype: "m.notice",
+              body: "You do not have the required power level in this room to create a bridge to a Discord channel.",
+          });
+      }
+
+      const prefix = "!discord ";
+      let command = "help";
+      let args = [];
+      if (event.content.body.length >= prefix.length) {
+          const allArgs = event.content.body.substring(prefix.length).split(" ");
+          if (allArgs.length && allArgs[0] !== "") {
+              command = allArgs[0];
+              allArgs.splice(0, 1);
+              args = allArgs;
+          }
+      }
+
+      if (command === "help" && args[0] === "bridge") {
+          const link = Util.GetBotLink(this.config);
+          this.bridge.getIntent().sendMessage(event.room_id, {
+              msgtype: "m.notice",
+              body: "How to bridge a Discord guild:\n" +
+              "1. Invite the bot to your Discord guild using this link: " + link + "\n" +
+              "2. Invite me to the matrix room you'd like to bridge\n" +
+              "3. Open the Discord channel you'd like to bridge in a web browser\n" +
+              "4. In the matrix room, send the message `!discord bridge <guild id> <channel id>` " +
+              "(without the backticks)\n" +
+              "   Note: The Guild ID and Channel ID can be retrieved from the URL in your web browser.\n" +
+              "   The URL is formatted as https://discordapp.com/channels/GUILD_ID/CHANNEL_ID\n" +
+              "5. Enjoy your new bridge!",
+          });
+      } else if (command === "bridge") {
+          if (context.rooms.remote) {
+              return this.bridge.getIntent().sendMessage(event.room_id, {
+                  msgtype: "m.notice",
+                  body: "This room is already bridged to a Discord guild.",
+              });
+          }
+
+          const minArgs = 2;
+          if (args.length < minArgs) {
+              return this.bridge.getIntent().sendMessage(event.room_id, {
+                  msgtype: "m.notice",
+                  body: "Invalid syntax. For more information try !discord help bridge",
+              });
+          }
+
+          const guildId = args[0];
+          const channelId = args[1];
+          try {
+              const discordResult = await this.discord.LookupRoom(guildId, channelId);
+              const channel = <Discord.TextChannel> discordResult.channel;
+
+              log.info("MatrixRoomHandler", `Bridging matrix room ${event.room_id} to ${guildId}/${channelId}`);
+              this.bridge.getIntent().sendMessage(event.room_id, {
+                  msgtype: "m.notice",
+                  body: "I'm asking permission from the guild administrators to make this bridge.",
+              });
+
+              await this.provisioner.AskBridgePermission(channel, event.sender);
+              await this.provisioner.BridgeMatrixRoom(channel, event.room_id);
```

Comment:
You don't need to await this one :)

Reply:
If we don't and bridging somehow fails then things get angry. This way it gets caught by the catch

Reply:
oh, hmm, it doesn't return a promise.

/me removes await